### PR TITLE
8292587: AArch64: Support SVE fabd instruction

### PR DESCRIPTION
**Pull request by shqking**
Scalar and NEON fabd instructions were initially supported in
JDK-8256318. In this patch, we support SVE fabd instruction [1] and add
one Jtreg test case as well.

With this patch, two instructions `fsub + fabs` would be combined into
one single `fabd` instruction.

```
  fsub    z16.s, z16.s, z17.s
  fabs    z16.s, p7/m, z16.s

  -->

  fabd    z16.s, p7/m, z16.s, z17.s
```

In the initial evaluation of JMH case, i.e.
FloatingScalarVectorAbsDiff.java, we found the performance uplift done
by this optimization was easily hidden by the heavy memory load/store
instructions. To avoid that, we updated the JMH case a bit, adding one
more group of subtraction and Math.abs operations in the loop body.

Here shows the data with the new JMH case on one 256-bit SVE machine. We
can observe about 39% and 35% improvements for the two functions
respectively.

```
Benchmark                                             Before    After  Units
FloatingScalarVectorAbsDiff.testVectorAbsDiffDouble  260.468  160.965  ns/op
FloatingScalarVectorAbsDiff.testVectorAbsDiffFloat   133.963   87.292  ns/op
```

Jtreg testing: tier1~3 passed on one NEON-only machine and one 256-bit SVE machine.

[1] https://developer.arm.com/documentation/ddi0596/2021-12/SVE-Instructions/FABD--Floating-point-absolute-difference--predicated--

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8292587](https://bugs.openjdk.org/browse/JDK-8292587): AArch64: Support SVE fabd instruction

### Reviewers
 * [Ningsheng Jian](https://openjdk.org/census#njian) (@nsjian - Committer) ⚠️ Review applies to [c6157252](https://git.openjdk.org/jdk/pull/10011/files/c6157252148f4e17863b713c1d7c9e05a4c5af6c)
 * [Fei Gao](https://openjdk.org/census#fgao) (@fg1417 - Author)
 * [Nick Gasson](https://openjdk.org/census#ngasson) (@nick-arm - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/10011/head:pull/10011` \
`$ git checkout pull/10011`

Update a local copy of the PR: \
`$ git checkout pull/10011` \
`$ git pull https://git.openjdk.org/jdk pull/10011/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 10011`

View PR using the GUI difftool: \
`$ git pr show -t 10011`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/10011.diff">https://git.openjdk.org/jdk/pull/10011.diff</a>

</details>

### CODE REVIEW

**Comment by shqking** [comment]
Ping? Can anyone help to review this patch? Thanks.

**Review by nsjian — APPROVED** [review]
Looks good.

**Comment by shqking** [comment]
Hi @nick-arm, could you help to review this patch when you have spare time? Thanks in advance.

File: test/hotspot/jtreg/compiler/vectorapi/VectorAbsDiffTest.java [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+            FloatVector av = FloatVector.fromArray(F_SPECIES, fa, i);
+            FloatVector bv = FloatVector.fromArray(F_SPECIES, fb, i);
+            av.sub(bv).lanewise(VectorOperators.ABS).intoArray(fr, i);
+        }
+    }
+
+    @Run(test = "testFloatAbsDiff")
+    public static void testFloatAbsDiff_runner() {
+        testFloatAbsDiff();
+        for (int i = 0; i < F_SPECIES.length(); i++) {
```
Comment (fg1417) [inline test/hotspot/jtreg/compiler/vectorapi/VectorAbsDiffTest.java:97]: I suppose it should be `for (int i = 0; i < LENGTH; i++) {` here. You can check all similar code lines here in the following  functions for verification.

Comment (shqking) [reply test/hotspot/jtreg/compiler/vectorapi/VectorAbsDiffTest.java]: Good catch! Updated it in the new revision. Thanks.

**Review by fg1417 — APPROVED** [review]

**Comment by TobiHartmann** [comment]
I tested this in our CI. All tests passed.

**Review by nick-arm — APPROVED** [review]

**Comment by shqking** [comment]
Thanks for your code reviews, @nick-arm @nsjian and @fg1417 .
Thanks for your testing, @TobiHartmann !

I don't think the GHA test failure is related to this patch.
Hence, I suppose we can integrate this PR now.

/integrate

**Pull request closed without merging**